src/Summoner/Project.hs

```diff
@@ -108,15 +107,16 @@ generateProject projectName Config{..} = do
 
     doGithubCommands :: Settings -> Bool -> IO ()
     doGithubCommands Settings{..} private = do
-        -- Create the repository on Github.
+        -- Create git repostitory and do a commit.
         "git" ["init"]
-        "hub" $ ["create", "-d", settingsDescription, settingsOwner <> "/" <> settingsRepo]
-             ++ ["-p" | private] -- creates private repository if asked so.
-        -- Make a commit and push it.
         "git" ["add", "."]
         "git" ["commit", "-m", "Create the project"]
-        "git" ["push", "-u", "origin", "master"]
-
+        unless noUpload ( do
+             -- Create private repository
```

Comment:
This comment is not always true. It just creates the repo. The private is depends only on `private` value.